rackcity/views/asset_views.py

```diff
@@ -335,13 +350,15 @@ def asset_modify(request):
             },
             status=HTTPStatus.BAD_REQUEST,
         )
-
-    failure_message = save_all_field_data(data, existing_asset, change_plan=change_plan)
+    print("Creating new asset cp",create_new_asset_cp)
+    (asset_cp,failure_message) = save_all_field_data(data, existing_asset, change_plan=change_plan,create_asset_cp = create_new_asset_cp)
     if failure_message:
         return JsonResponse(
             {"failure_message": Status.MODIFY_ERROR.value + failure_message},
             status=HTTPStatus.BAD_REQUEST,
         )
+    if asset_cp:
+        existing_asset = asset_cp
```

Comment:
need to get the new assetCP now because  the rest of the logic assumes that it is getting an assetcp

Reply:
see comment above about that method change